src/driver/xen.rs

```diff
@@ -164,6 +184,86 @@ impl Introspectable for Xen {
         }))
     }
 
+    fn listen(&mut self, timeout: u32) -> Result<Option<Event>, Box<dyn Error>> {
+        let fd = self.xev.xenevtchn_fd()?;
+        let fd_struct = PollFd::new(fd, PollFlags::POLLIN | PollFlags::POLLERR);
+        let mut fds = [fd_struct];
+        let mut vcpu: u16 = 0;
+        let mut event_type = unsafe { mem::MaybeUninit::<EventType>::zeroed().assume_init() };
+        let poll_result = poll(&mut fds, timeout.try_into().unwrap()).unwrap();
+        let mut pending_event_port = -1;
+        if poll_result == 1 {
+            pending_event_port = self.xev.xenevtchn_pending()?;
+            if pending_event_port != -1 {
+                self.xev
+                    .xenevtchn_unmask(pending_event_port.try_into().unwrap())?;
+            }
+        }
+        let back_ring_ptr = &mut self.back_ring;
+        let mut flag = false;
```

Comment:
`flag` doesn't mean anything.

Reply:
actually, you don't need it. your `if` statement can return a value, which is your Event, or None.
just call notify, and then Ok on the value returned by the if